GitUI/UserControls/BlameControl.cs

```diff
@@ -599,46 +601,47 @@ private void copyCommitHashToClipboardToolStripMenuItem_Click(object sender, Eve
             CopyToClipboard(c => c.ObjectId.ToString());
         }
 
-        private bool TryGetSelectedRevision([NotNullWhen(returnValue: true)] out GitRevision? selectedRevision)
+        private bool TryGetSelectedRevision([NotNullWhen(returnValue: true)] out (GitRevision? selectedRevision, string? filename) blameInfo)
         {
             var blameCommit = GetBlameCommit();
             if (blameCommit is null)
             {
-                selectedRevision = null;
+                blameInfo = (null, null);
                 return false;
             }
 
-            selectedRevision = _revGrid?.GetRevision(blameCommit.ObjectId);
-            return selectedRevision is not null;
+            blameInfo = (_revGrid?.GetRevision(blameCommit.ObjectId), blameCommit.FileName);
+            return blameInfo.selectedRevision is not null;
         }
 
         private void blameRevisionToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            if (!TryGetSelectedRevision(out var selectedRevision))
+            if (!TryGetSelectedRevision(out var blameInfo))
             {
                 return;
             }
 
-            BlameRevision(selectedRevision.ObjectId);
+            BlameRevision(blameInfo.selectedRevision.ObjectId, blameInfo.filename);
         }
 
         private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            if (!TryGetSelectedRevision(out var selectedRevision))
+            if (!TryGetSelectedRevision(out var blameInfo))
             {
                 return;
             }
 
             // Try get actual parent revision, get popup if it does not exist.
             // (The menu should be disabled if previous is not in grid).
-            GitRevision? revision = _revGrid?.GetActualRevision(selectedRevision);
-            BlameRevision(revision?.FirstParentId);
+            GitRevision? revision = _revGrid?.GetActualRevision(blameInfo.selectedRevision);
+            BlameRevision(revision?.FirstParentId, blameInfo.filename);
```

Comment:
Can the filename be changed here already?
Not a blocker for the PR, but it warrants a comment.

Reply:
I meant if the info.fileName was already changed and be incorrect here.

> I thought that changing the method provided a more understandable contract.

I thought of that too, but will approve just if there are added comments.

> and that let me see that there is a regression in the way we display file path in the blame gutter 😢 -- a problem of UNIX vs Windows path format

Do you want to display with Windows paths?
Presentation is generally in Unix paths so way different here?

Reply:
> I meant if the info.fileName was already changed and be incorrect here.
> [...] but will approve just if there are added comments

I'm happy to put comments but I don't really get what you are asking me.

If the file has been renamed on the parent commit also (so 2 consecutive renames)?
Yes, we will end up with the same behavior than previously because we don't have the good path. But we can't do better with the information we got.

> Do you want to display with Windows paths? Presentation is generally in Unix paths so way different here?

No, I don't want to change the way the path are displayed. That's just that previously the path was displayed only if it is different than the one of the file you are blaming to have less information to handle as a user.
Like that:
![d](https://user-images.githubusercontent.com/460196/78377835-708a9880-75d0-11ea-887c-0181db1e548f.png)

But for that you have to compare paths ( https://github.com/gitextensions/gitextensions/blob/master/GitUI/UserControls/BlameControl.cs#L442 ). There is just a behavior change where now, the `filename` received is in Windows format whereas it was in Unix format before to have the comparison working.
I have already the one line fix. Just have to push it...